src/Worker.Extensions.DurableTask/TaskOrchestrationContextExtensionMethods.cs

```diff
@@ -20,14 +24,47 @@ public static class TaskOrchestrationContextExtensionMethods
     /// <param name="context">The task orchestration context.</param>
     /// <param name="request">The DurableHttpRequest used to make the HTTP call.</param>
     /// <returns>DurableHttpResponse</returns>
-    public static Task<DurableHttpResponse> CallHttpAsync(this TaskOrchestrationContext context, DurableHttpRequest request)
+    public static async Task<DurableHttpResponse> CallHttpAsync(this TaskOrchestrationContext context, DurableHttpRequest request)
     {
         if (context is null)
         {
             throw new ArgumentNullException(nameof(context));
         }
 
-        return context.CallActivityAsync<DurableHttpResponse>(Constants.HttpTaskActivityReservedName, request);
+        DurableHttpResponse response = await context.CallActivityAsync<DurableHttpResponse>(Constants.HttpTaskActivityReservedName, request);
+        
+        while (response.StatusCode == HttpStatusCode.Accepted && request.AsynchronousPatternEnabled )
+        {
+            // If Headers is null or missing, we can't poll the Location URL, so return the response.
+            if (response.Headers is null)
+            {
+                break;
+            }
+
+            var headersDictionary = new Dictionary<string, StringValues>(
+                       response.Headers!,
+                       StringComparer.OrdinalIgnoreCase);
+
+            DateTime fireAt = default(DateTime);
+
+            if (headersDictionary.TryGetValue("Retry-After", out StringValues retryAfter))
+            {
+                fireAt = context.CurrentUtcDateTime.AddSeconds(int.Parse(retryAfter));
+            }
+            else
+            {
+                // TODO: use the configuration DefaultAsyncRequestSleepTimeMilliseconds from durabletaskextension
```

Comment:
The context here is from [TaskOrchestrationContext](https://github.com/microsoft/durabletask-dotnet/blob/main/src/Abstractions/TaskOrchestrationContext.cs), and it doesn't have property saving config from DurableExtension. We can do this but that might require quite a lot changes, including DurableTask.Abstractions. Thus I am not sure if we should do that, or if there is any other easy way to do that.

Reply:
Hmm. I'm not sure the right approach for this. One option would be to introduce a generic property bag (`Dictionary<string, object>`) in `TaskOrchestrationContext`, and have that be populated using data from the trigger context that we get from the WebJobs extension. For example, adding a a key named `df.http.defaultAsyncRequestSleepTimeMilliseconds` and have the value come from host.json. We still have to add the property bag to the DurableTask.Abstractions, but we only need to make that one change and we can easily add other properties to the dictionary in the future without changing DurableTask.Abstractions a second time. Let me know if I can help explain this idea better.